bluepysnap/_plotting.py

```diff
@@ -0,0 +1,363 @@
+# Copyright (c) 2020, EPFL/Blue Brain Project
+
+# This file is part of BlueBrain SNAP library <https://github.com/BlueBrain/snap>
+
+# This library is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License version 3.0 as published
+# by the Free Software Foundation.
+
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""Plotting module for the different snap objects."""
+import logging
+import numpy as np
+import pandas as pd
+
+from bluepysnap.exceptions import BluepySnapError
+from bluepysnap.sonata_constants import Node
+from bluepysnap.utils import roundrobin
+
+L = logging.getLogger(__name__)
+
+
+def _get_pyplot():
+    try:
+        import matplotlib.pyplot as plt
+    except ImportError as e:
+        msg = (
+            "Bluepysnap requirements are not installed.\n"
+            "Please pip install as follows:\n"
+            "  pip install bluepysnap[plots] --upgrade"
+        )
+        raise ImportError(str(e) + "\n\n" + msg)
+    return plt
+
+
+def spikes_firing_rate_histogram(filtered_report, time_binsize=None, ax=None):  # pragma: no cover
+    """Spike firing rate histogram.
+
+    This plot shows the number of nodes firing during a range of time.
+
+    Args:
+        time_binsize(int): bin size (milliseconds)
```

Comment:
What does the default of `None` mean? Include `None` in the type. Is a float ever acceptable?

Reply:
Totally acceptable. 
I changed it for None/int/float and added a comment about the None. 
I added a raise if time_binsize <= 0.